tests/Microsoft.Identity.Test.Unit/TelemetryTests/OTelInstrumentationTests.cs

```diff
@@ -0,0 +1,145 @@
+{
+    [TestClass]
+    [DeploymentItem(@"Resources\testCert.crtfile")]
+    [DeploymentItem(@"Resources\RSATestCertDotNet.pfx")]
+    public class OTelInstrumentationTests : TestBase
+    {
+        private MockHttpAndServiceBundle _harness;
+        private ConfidentialClientApplication _cca;
+        private static MeterProvider s_meterProvider;
+        private static TracerProvider s_activityProvider;
+        private readonly List<Metric> _exportedMetrics = new();
+        private readonly List<Activity> _exportedActivities = new();
+
+        [TestCleanup]
+        public override void TestCleanup()
+        {
+            s_meterProvider?.Dispose();
+            s_activityProvider?.Dispose();
+
+            _exportedMetrics.Clear();
+            _exportedActivities.Clear();
+
+            base.TestCleanup();
+        }
+
+        [TestInitialize]
+        public override void TestInitialize()
+        {
+            base.TestInitialize();
+            _harness = CreateTestHarness();
+
+            s_meterProvider = Sdk.CreateMeterProviderBuilder()
+                .AddMeter(OtelInstrumentation.MeterName)
+                .AddInMemoryExporter(_exportedMetrics)
+                .Build();
+
+            s_activityProvider = Sdk.CreateTracerProviderBuilder()
+                .AddSource(OtelInstrumentation.ActivitySourceName)
+                .AddInMemoryExporter(_exportedActivities)
+                .Build();
+        }
+
+        [TestMethod]
+        public async Task AcquireTokenOTelTestAsync()
+        {
+            using (_harness = CreateTestHarness())
+            {
+                CreateApplication();
+                await AcquireTokenSuccessAsync().ConfigureAwait(false);
+                await AcquireTokenMsalServiceExceptionAsync().ConfigureAwait(false);
+                await AcquireTokenMsalClientExceptionAsync().ConfigureAwait(false);
+
+                s_meterProvider.ForceFlush();
+                s_activityProvider.ForceFlush();
+                OTelInstrumentationUtil.VerifyMetrics(5, _exportedMetrics);
+                OTelInstrumentationUtil.VerifyActivity(19, _exportedActivities);
+            }
+        }
+
+        private async Task AcquireTokenSuccessAsync()
+        {
+            _harness.HttpManager.AddInstanceDiscoveryMockHandler();
+            _harness.HttpManager.AddMockHandlerSuccessfulClientCredentialTokenResponseMessage();
+
+            // Acquire token for client with scope
+            var result = await _cca.AcquireTokenForClient(TestConstants.s_scope)
+                .ExecuteAsync(CancellationToken.None).ConfigureAwait(false);
+            Assert.IsNotNull(result);
+
+            // Acquire token silently
+            var account = (await _cca.GetAccountsAsync().ConfigureAwait(false)).Single();
+            result = await _cca.AcquireTokenSilent(TestConstants.s_scope, account)
+                .ExecuteAsync().ConfigureAwait(false);
+            Assert.IsNotNull(result);
```

Comment:
did we miss checking any successful metric here?

Reply:
Since otel metrics measures aggregation, I have coupled the success and failure tests and verifying the metrics together. So this is a method to only acquire token and assert that we get a successful response. Metrics is verified at line 76.